File: C++/Baekjoon/Problem10799.cpp
#include <iostream>
#include <string>
using namespace std;

int main()
{
    int cnt=0;
    int res=0;
    string s;
    cin>>s;
    for(int i=0; i<=s.length(); i++)
    {
        if(s[i]=='(')
        {
            if(s[i+1]==')')
            {
                res+=cnt;
                i=i+1;
                continue;
            }
            cnt++;
        }
        else if(s[i]==')')
        {
            res++;
            cnt--;
        }
    }
    cout<<res;
}

File: Python/Baekjoon/Problem10799.py
res = 0
cnt = 0
s = input()
for i in range(len(s)):
    if s[i] == "(":
        if s[i + 1] == ")":
            res += cnt
            i = i + 1
            continue
        cnt += 1
    if s[i - 1] != "(" and s[i] == ")":
        res += 1
        cnt -= 1

print(res)
